### ISSUE TITLE
notebooks: cannot assign writer privilege to group members

### ISSUE THREAD

**Issue by litmus-ritten** [tloncorp/landscape-apps#1572]
May be related to #1504. I find that I cannot assign the writer privilege for users to allow them to post to nominated notebooks. In the Group admin interface, the dropdown chevron menu presents notebooks to assign writer status to on a per-user basis, however clicking on any option just produces a spinner next to each option for a fraction of a second and closes the menu. There is also an option to promote a user to admin but I have not tried that.

**To Reproduce**
Steps to reproduce the behaviour:
1. Have a notebook that only administrators and writers can post to.
2. Nominate a group member to be a writer for that notebook.
3. Note that nothing happens.

**Expected behaviour**
Users can have writer status assigned and revoked from this interface.

**Additional context:**
- Ship type: planet
- Instance type: self-hosted CLI
- OS/browser: Vivaldi on Linux

**Comment by jamesacklin** [comment]
Additional functionality to be addressed as part of #539.

**Comment by arthyn** [comment]
@litmus-ritten I'm not sure what screen you're talking about here where you can see members to promote as writers?

**Comment by litmus-ritten** [comment]
@arthyn See the membership list in a group where there exists notebooks with restricted write access. The chevron menu will present a list of notebooks write privileges can be assigned for.

![Screenshot from 2023-01-07 13-19-56](https://user-images.githubusercontent.com/62781197/211127364-22b086d0-ea01-4a74-8c61-1e45ac5f5a87.png)

**Comment by arthyn** [comment]
I don't think this is a feature in our apps. Is this escape? @litmus-ritten

**Comment by litmus-ritten** [comment]
@arthyn No, this is Groups, with a desk hash reported in the corresponding Garden tile ending in `3shua`. The colours look different because I'm injecting custom CSS which replaces the dark theme colours, otherwise see the following `+vats` output:

```
%groups
  /sys/kelvin:      [%zuse 417]
  base hash:        0vs.n20oi.r18b8.ruh8k.250eh.iummk.p8udg.8l23j.b9628.n3sao.j2gsd
  %cz hash:         0vs.n20oi.r18b8.ruh8k.250eh.iummk.p8udg.8l23j.b9628.n3sao.j2gsd
  app status:       running
  force on:         ~
  force off:        ~
  publishing ship:  ~
  updates:          remote
  source ship:      ~sogryp-dister-dozzod-dozzod
  source desk:      %groups
  source aeon:      6
  kids desk:        ~
  pending updates:  ~
::
```

This menu probably won't display anything interesting unless you have a group that contains non-admin members and one or more notebooks with _'members can view'_ privileges.

**Comment by jamesacklin** [comment]
This was fixed in v3.4.0; we now let you add members to a role and then set the notebook read/write permissions by role.

**Comment by litmus-ritten** [comment]
@jamesacklin  If I recall correctly, someone pointed out in chat that the time required to set permissions scales viciously with the number of members in a group. I just get an indefinite spinner when I try to set someone as having an authorship role in a group with 115 members. If said commentator is correct, this may make setting permissions infeasible for moderately large groups.
